site/src/pages/ManagementSettingsPage/CustomRolesPage/CreateEditRolePageView.tsx

```diff
@@ -0,0 +1,190 @@
+              })}
+              fullWidth
+              label="Display Name"
+            />
+            <ActionCheckboxes
+              permissions={role?.organization_permissions || []}
+              form={form}
+            />
+          </FormFields>
+        </FormSection>
+        <FormFooter
+          onCancel={onCancel}
+          isLoading={isLoading}
+          submitLabel={role !== undefined ? "Save" : "Create Role"}
+        />
+      </HorizontalForm>
+    </>
+  );
+};
+
+interface ActionCheckboxesProps {
+  permissions: readonly Permission[] | undefined;
+  form: ReturnType<typeof useFormik<Role>> & { values: Role };
+}
+
+const ActionCheckboxes: FC<ActionCheckboxesProps> = ({ permissions, form }) => {
+  const [checkedActions, setCheckActions] = useState(permissions);
+
+  const handleCheckChange = async (
+    e: ChangeEvent<HTMLInputElement>,
+    form: ReturnType<typeof useFormik<Role>> & { values: Role },
+  ) => {
+    const { name, checked } = e.currentTarget;
+    const [resource_type, action] = name.split(":");
+
+    const newPermissions = checked
+      ? [
+          ...(checkedActions ?? []),
+          {
+            negate: false,
+            resource_type: resource_type as RBACResource,
+            action: action as RBACAction,
+          },
+        ]
+      : checkedActions?.filter(
+          (p) => p.resource_type !== resource_type || p.action !== action,
+        );
+
+    setCheckActions(newPermissions);
+    await form.setFieldValue("organization_permissions", newPermissions);
+  };
+
+  return (
+    <TableContainer>
+      <Table>
+        <TableBody>
+          {Object.entries(RBACResourceActions).map(([resourceKey, value]) => {
+            return (
+              <TableRow key={resourceKey}>
+                <TableCell>
+                  <li key={resourceKey} css={styles.checkBoxes}>
+                    {resourceKey}
+                    <ul css={styles.checkBoxes}>
+                      {Object.entries(value).map(([actionKey, value]) => {
+                        return (
+                          <li key={actionKey}>
+                            <span css={styles.actionText}>
+                              <Checkbox
+                                name={`${resourceKey}:${actionKey}`}
+                                checked={
+                                  checkedActions?.some(
+                                    (p) =>
+                                      p.resource_type === resourceKey &&
+                                      (p.action.toString() === "*" ||
+                                        p.action === actionKey),
+                                  ) || false
+                                }
+                                onChange={(e) => handleCheckChange(e, form)}
+                              />
+                              {actionKey}
+                            </span>{" "}
+                            -{" "}
```

Comment:
```suggestion
                            &ndash;{" "}
```

super nit picky, but an n-dash feels more appropriate here than a hyphen.

Reply:
got it, btw, this was meant to be temporary as I wanted to create a cleaner separation of the action and action description.